clevertap-core/src/main/java/com/clevertap/android/sdk/Utils.java

```diff
@@ -477,6 +478,27 @@ private static boolean checkForExoPlayer() {
         return exoPlayerPresent;
     }
 
+    /**
+     * Method to check whether app has upgraded to Firebase Cloud Messaging v22.0.0 or greater.
+     *
+     * @return boolean - true/false depending on app's availability of FirebaseInstanceId in FirebaseMessaging
+     *                   dependencies
+     */
+    @SuppressWarnings("rawtypes")
+    private static boolean checkForFirebaseInstanceId(){
+        boolean isFirebaseInstanceIdPresent = false;
+        Class classname;
+        try {
+            classname = Class.forName("com.google.firebase.iid.FirebaseInstanceId");
+            isFirebaseInstanceIdPresent = true;
+            Logger.d("Firebase Instance Id is available." + classname);
+        }catch (Throwable throwable){
+            Logger.d("It looks like you're using FirebaseMessaging dependency v22.0.0. " +
+                    "Ensure your app's version of FirebaseMessaging is less than 20.3.0.");
```

Comment:
@william-ct Be specific in the logging here. "Ensure your app's version of FCM is v20.2.4"